src/connector/src/parser/parquet_parser.rs

```diff
@@ -0,0 +1,177 @@
+                        let rw_data_type = &source_column.data_type;
+                        let rw_column_name = &source_column.name;
+                        if let Some(parquet_column) = record_batch.column_by_name(rw_column_name) {
+                            let arrow_field =
+                                IcebergArrowConvert.to_arrow_field(rw_column_name, rw_data_type)?;
+                            let converted_arrow_data_type: &arrow_schema_iceberg::DataType =
+                                arrow_field.data_type();
+                            if converted_arrow_data_type == parquet_column.data_type() {
+                                let array_impl = IcebergArrowConvert
+                                    .array_from_arrow_array(&arrow_field, parquet_column)?;
+                                let column = Arc::new(array_impl);
+                                chunk_columns.push(column);
+                            } else {
+                                // data type mismatch, this column is set to null.
+                                let mut array_builder =
+                                    ArrayBuilderImpl::with_type(size, rw_data_type.clone());
+
+                                array_builder.append_n_null(record_batch.num_rows());
+                                let res = array_builder.finish();
+                                let column = Arc::new(res);
+                                chunk_columns.push(column);
+                            }
+                        } else {
+                            // For columns defined in the source schema but not present in the Parquet file, null values are filled in.
+                            let mut array_builder =
+                                ArrayBuilderImpl::with_type(size, rw_data_type.clone());
+
+                            array_builder.append_n_null(record_batch.num_rows());
+                            let res = array_builder.finish();
+                            let column = Arc::new(res);
+                            chunk_columns.push(column);
+                        }
+                    }
+                    // handle hidden columns, for file source, the hidden columns are only `Offset` and `Filename`
+                    true => {
+                        if let Some(additional_column_type) =
+                            &source_column.additional_column.column_type
+                        {
+                            match additional_column_type{
+                                risingwave_pb::plan_common::additional_column::ColumnType::Offset(_) =>{
+                                    let mut array_builder =
+                                    ArrayBuilderImpl::with_type(size, source_column.data_type.clone());
+                                    let datum: Datum =  Some(ScalarImpl::Utf8(offset.to_string().into()));
+                                    array_builder.append_n(record_batch.num_rows(), datum);
```

Comment:
This assigns the same row offset for all rows in a record batch, which seems incorrect to me. We may need a test case to test the correctness of seek by row offset and verify the behavior.